lte/gateway/c/oai/tasks/mme_app/mme_app_bearer.c

```diff
@@ -3716,6 +3726,335 @@ ue_mm_context_t* mme_app_get_ue_context_for_timer(
+  MessageDef* message_p                = NULL;
+
+  if (!e_rab_modification_ind->e_rab_to_be_modified_list.no_of_items) {
+    OAILOG_NOTICE(
+        LOG_MME_APP,
+        "S1AP E-RAB_MODIFICATION_IND no e-rab to be modified for "
+        "mme_ue_s1ap_id: " MME_UE_S1AP_ID_FMT "\n",
+        e_rab_modification_ind->mme_ue_s1ap_id);
+    OAILOG_FUNC_OUT(LOG_MME_APP);
+  }
+
+  // ue_context_p =
+  // mme_ue_session_pool_exists_mme_ue_s1ap_id(&mme_app_desc.mme_ue_session_pools,e_rab_modification_ind->mme_ue_s1ap_id);
+
+  ue_context_p = mme_ue_context_exists_mme_ue_s1ap_id(
+      e_rab_modification_ind->mme_ue_s1ap_id);
+
+  if (ue_context_p == NULL) {
+    OAILOG_DEBUG(
+        LOG_MME_APP,
+        "We didn't find this mme_ue_s1ap_id in list of UE: " MME_UE_S1AP_ID_FMT
+        "\n",
+        e_rab_modification_ind->mme_ue_s1ap_id);
+    OAILOG_FUNC_OUT(LOG_MME_APP);
+  }
+  /* check ETSI TS 136 413 V15.5.0 8.2.4.4 Abnormal Conditions
+   * If the E-RAB MODIFICATION INDICATION message does not contain all the
+   * E-RABs previously included in the UE Context, the MME shall trigger the UE
+   * Context Release procedure.
+   */
+
+  for (int i = 0;
+       i < e_rab_modification_ind->e_rab_to_be_modified_list.no_of_items; i++) {
+    e_rab_id_t e_rab_id =
+        e_rab_modification_ind->e_rab_to_be_modified_list.item[i].e_rab_id;
+    // bearer_context_new_t* bearer_context = NULL;
+
+    bearer_context_t* bearer_context =
+        mme_app_get_bearer_context(ue_context_p, (ebi_t) e_rab_id);
+
+    if (!bearer_context) {
+      OAILOG_NOTICE(
+          LOG_MME_APP,
+          "S1AP E-RAB_MODIFICATION_IND no e-rab %d to be modified for "
+          "mme_ue_s1ap_id: " MME_UE_S1AP_ID_FMT " -> Releasing context...\n",
+          e_rab_modification_ind->e_rab_to_be_modified_list.item[i].e_rab_id,
+          e_rab_modification_ind->mme_ue_s1ap_id);
+
+      ue_context_p = mme_ue_context_exists_mme_ue_s1ap_id(
+          e_rab_modification_ind->mme_ue_s1ap_id);
+      ue_context_p->ue_context_rel_cause = S1AP_RADIO_UNKNOWN_E_RAB_ID;
+      mme_app_itti_ue_context_release(
+          ue_context_p, ue_context_p->ue_context_rel_cause);
+    }
+    /*
+     * If the E-RAB MODIFICATION INDICATION message contains several E-RAB ID
+     * IEs set to the same value, the MME shall trigger the UE Context Release
+     * procedure.
+     */
+    for (int j = 0; j < i; j++) {
+      if (e_rab_modification_ind->e_rab_to_be_modified_list.item[i].e_rab_id ==
+          e_rab_modification_ind->e_rab_to_be_modified_list.item[j].e_rab_id) {
+        ue_context_p->ue_context_rel_cause = S1AP_RADIO_MULTIPLE_E_RAB_ID;
+        mme_app_itti_ue_context_release(
+            ue_context_p, ue_context_p->ue_context_rel_cause);
+        OAILOG_FUNC_OUT(LOG_MME_APP);
+      }
+    }
+  }
+
+  for (int i = 0;
+       i < e_rab_modification_ind->e_rab_not_to_be_modified_list.no_of_items;
+       i++) {
+    bearer_context_new_t* bearer_context = NULL;
```

Comment:
When is it modified ?

Reply:
The test should be on ue_context_p not bearer_context: (my code release had this issue I imagine)
You can remove bearer_context from here

    bearer_context_new_t* bearer_context = NULL;
    mme_app_get_bearer_context(
        ue_context_p,
        e_rab_modification_ind->e_rab_not_to_be_modified_list.item[i].e_rab_id);
    if (!bearer_context) {

Reply:
Resolved